src/solvers/iterators.jl

```diff
@@ -0,0 +1,104 @@
+#
+# SweepIterator
+#
+
+mutable struct SweepIterator
+  sweep_kws
```

Comment:
Maybe this could be named `each_sweep_kwargs` or `sweep_kwargs_iterator` to indicate that it itself is an iterator that returns the keyword arguments for the sweep at each iteration (rather than just the keyword arguments of the latest sweep).